Mathlib/GroupTheory/GroupAction/FixedPoints.lean

```diff
@@ -0,0 +1,322 @@
+  do not commute from the disjointness of the `movedBy` set and its image by the second group
+  element, which is a property used in the proof of Rubin's theorem.
+-/
+
+namespace MulAction
+open Pointwise
+
+variable {α : Type*}
+variable {G : Type*} [Group G] [MulAction G α]
+variable {M : Type*} [Monoid M] [MulAction M α]
+
+
+section FixedPoints
+
+variable (α) in
+/-- In a multiplicative group action, the points fixed by `g` are also fixed by `g⁻¹` -/
+@[to_additive (attr := simp)
+  "In an additive group action, the points fixed by `g` are also fixed by `g⁻¹`"]
+theorem fixedBy_inv_eq_fixedBy (g : G) : fixedBy α g⁻¹ = fixedBy α g := by
+  ext x
+  rw [mem_fixedBy, mem_fixedBy, inv_smul_eq_iff, eq_comm]
+
+@[to_additive]
+theorem smul_mem_fixedBy_iff_mem_fixedBy {a : α} {g : G} :
+    g • a ∈ fixedBy α g ↔ a ∈ fixedBy α g := by
+  rw [mem_fixedBy, smul_left_cancel_iff]
+  rfl
+
+theorem minimalPeriod_eq_one_of_fixedBy {a : α} {g : G} (a_in_fixedBy : a ∈ fixedBy α g) :
+    Function.minimalPeriod (fun x => g • x) a = 1 := by
+  rwa [Function.minimalPeriod_eq_one_iff_isFixedPt]
+
+section AddAction.minimalPeriod_eq_one_of_fixedBy
+variable {G : Type*} [AddGroup G] [AddAction G α]
+
+theorem _root_.AddAction.minimalPeriod_eq_one_of_fixedBy {a : α} {g : G}
+    (a_in_fixedBy : a ∈ AddAction.fixedBy α g) :
+    Function.minimalPeriod (fun x => g +ᵥ x) a = 1 := by
+  rwa [Function.minimalPeriod_eq_one_iff_isFixedPt]
+
+end AddAction.minimalPeriod_eq_one_of_fixedBy
+
+attribute [to_additive existing AddAction.minimalPeriod_eq_one_of_fixedBy]
+  minimalPeriod_eq_one_of_fixedBy
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_subset_fixedBy_zpow (g : G) (j : ℤ) :
+    fixedBy α g ⊆ fixedBy α (g ^ j) := by
+  intro a a_in_fixedBy
+  rw [mem_fixedBy, zpow_smul_eq_iff_minimalPeriod_dvd,
+    minimalPeriod_eq_one_of_fixedBy a_in_fixedBy, Nat.cast_one]
+  exact one_dvd j
+
+variable (M α) in
+@[to_additive (attr := simp)]
+theorem fixedBy_one_eq_univ : fixedBy α (1 : M) = Set.univ :=
+  Set.eq_univ_iff_forall.mpr (fun a => one_smul M a)
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_mul (m₁ m₂ : M) : fixedBy α m₁ ∩ fixedBy α m₂ ⊆ fixedBy α (m₁ * m₂) := by
+  intro a ⟨h₁, h₂⟩
+  rw [mem_fixedBy, mul_smul, h₂, h₁]
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_conj (g h : G) :
+    fixedBy α (h * g * h⁻¹) = (fun a => h⁻¹ • a) ⁻¹' fixedBy α g := by
+  ext a
+  rw [Set.mem_preimage, mem_fixedBy, mem_fixedBy]
+  repeat rw [mul_smul]
+  exact smul_eq_iff_eq_inv_smul h
+
+end FixedPoints
+
+section MovedBy
+
+/-!
+## Moved points
+
+`MulAction.movedBy` and `AddAction.movedBy` describe the sets of points `a : α` for which
+`g • a ≠ a` (resp. `g +ᵥ a ≠ a`).
+-/
+
+variable (α) in
+/-- The set of points moved by an element of the action. -/
+@[to_additive "The set of points moved by an element of the action."]
+def movedBy (m : M) : Set α := { a : α | m • a ≠ a }
```

Comment:
Is just writing `(fixedBy α m)ᶜ` everywhere an option, or are some things too painful?

Reply:
There are a few works out there that chain operations with this structure, which they usually call the "support" of the action; this is the case in Rubin's theorem. I don't mind changing this to an `abbrev` or `notation`, though, and that would probably allow me to drop a few theorems